pyccel/codegen/printing/ccode.py

```diff
@@ -1134,35 +1181,51 @@ def formatted_args_to_printf(args_format, args, end):
             code += formatted_args_to_printf(args_format, args, end)
         return code
 
-    def find_in_dtype_registry(self, dtype):
+    def find_in_type_registry(self, dtype):
         """
-        Find the corresponding C dtype in the dtype_registry.
+        Find the corresponding C dtype in the dtype_registry or the STC container type.
 
-        Find the corresponding C dtype in the dtype_registry.
-        Raise PYCCEL_RESTRICTION_TODO if not found.
+        This function searches for the corresponding C data type in the `dtype_registry`.
+        If the provided dtype is a container (like `HomogeneousSetType` or
+        `HomogeneousListType`), it identifies the scalar type of the container and returns
+        the appropriate type for the `STC` container.
+        It raises a `PYCCEL_RESTRICTION_TODO` error if the dtype is not found in the registry.
 
         Parameters
         ----------
-        dtype : DataType
-            The data type of the expression.
+        dtype : DataType or ContainerType
+            The data type of the expression. This can be a fixed-size numeric type,
+            a primitive type, or a container type.
 
         Returns
         -------
         str
-            The code which declares the datatype in C.
+            The code which declares the data type in C or the corresponding `STC` container
+            type.
+
+        Raises
+        ------
+        PYCCEL_RESTRICTION_TODO
+            If the dtype is not found in the dtype_registry.
         """
         if isinstance(dtype, FixedSizeNumericType):
             primitive_type = dtype.primitive_type
             if isinstance(primitive_type, PrimitiveComplexType):
                 self.add_import(c_imports['complex'])
-                return f'{self.find_in_dtype_registry(dtype.element_type)} complex'
+                return f'{self.find_in_type_registry(dtype.element_type)} complex'
             elif isinstance(primitive_type, PrimitiveIntegerType):
                 self.add_import(c_imports['stdint'])
             elif isinstance(dtype, PythonNativeBool):
                 self.add_import(c_imports['stdbool'])
                 return 'bool'
 
             key = (primitive_type, dtype.precision)
+        elif isinstance(dtype, (HomogeneousSetType, HomogeneousListType)):
+            container_type = 'hset_' if dtype.name == 'set' else 'vec_'
+            element_type = self.find_in_type_registry(dtype.element_type)
+            i_type = container_type + element_type.replace(' ', '_')
+            self.add_import(Import(f'stc/{i_type}/{element_type}', Module(f'stc/{i_type}', (), ())))
```

Comment:
When initializing an object of class `Import`, it runs an assertion that expects a `Module` object.
```python
assert isinstance(i, (AsName, Module))
```

Reply:
Ah I didn't realise this. I was actually hoping that we could use the target parameter to avoid fiddling around with strings.

I have had to make this change in order to support STC arrays via cspan (the import is the same in this case). I am tempted to incorporate the change in this PR but I think it might be better to leave it as an improvement in a later PR (at least so I don't review too much of my own code).

In case you're interested this is what I had in mind:
https://github.com/pyccel/pyccel/blob/8c04f46c13ec01074d60f48413ae1137a5518d90/pyccel/codegen/printing/ccode.py#L1228

Reply:
Looks better and more readable,I wish we had taken this approach earlier.
which PR are you using this?

Reply:
I haven't opened the PR yet. I think I will open one just for this once this PR is merged. The branch is `ebourne_stc_arrays`